audb/core/publish.py

```diff
@@ -133,15 +133,23 @@ def _find_attachments(
     ):
         # use one archive per attachment ID
         path = db.attachments[attachment_id].path
-        checksum = utils.md5(audeer.path(db_root, path))
-        if path not in deps or checksum != deps.checksum(path):
-            deps._add_attachment(
-                file=path,
-                version=version,
-                archive=attachment_id,
-                checksum=checksum,
-            )
-            attachment_ids.append(attachment_id)
+        if not os.path.exists(audeer.path(db_root, path)):
+            if path not in deps:
+                # Force raising error
```

Comment:
Could you please be a little bit more verbose what we do here?

Reply:
Ah now I get it - this covers the case of a new attachment that does not exist on disk.

Reply:
Yes, but it could indeed be more verbose. I changed to

```python
                # Raise FileNotFoundError
                #   
                # Attachment is not in deps,
                # but its path does not exist on disk neither.
                # We call its `files` property
                # which raises a FileNotFoundError in this case
```

Reply:
Thanks!